app/models/manageiq/providers/ansible_tower/shared/automation_manager/credential.rb

```diff
@@ -1,75 +1,18 @@
 module ManageIQ::Providers::AnsibleTower::Shared::AutomationManager::Credential
   extend ActiveSupport::Concern
 
+  include ProviderObjectMixin
+
   module ClassMethods
-    def create_in_provider(manager_id, params)
-      manager = ExtManagementSystem.find(manager_id)
-      credential = manager.with_provider_connection do |connection|
-        connection.api.credentials.create!(params)
+    def provider_collection(manager)
```

Comment:
That's a good idea and I am happy to implement it.

However, I am not sure how useful it is in this case. Because the value with the `with_*` pattern depends on wether the returned object needs and supports the `object.release`.

Here we are only using the Tower specific API. I don't think it has this need.  Let me know if I miss anything.